h2/src/main/org/h2/value/ValueLob.java

```diff
@@ -66,6 +68,87 @@ static InputStream rangeInputStream(InputStream inputStream, long oneBasedOffset
         }
     }
 
+    /**
+     * Compares LOBs of the same type.
+     *
+     * @param valueType type
+     * @param v1 first LOB value
+     * @param v2 second LOB value
+     * @return result of comparison
+     */
+    static int compare(int valueType, Value v1, Value v2) {
+        long prec1 = v1.getPrecision(), prec2 = v2.getPrecision();
+        if (Math.max(prec1, prec2) <= BLOCK_COMPARISON_SIZE) {
```

Comment:
there is a constant below which the LOB data is stored inline that I think should be used here

Reply:
database.getMaxLengthInplaceLob

Reply:
I see only `DataHandler.getMaxLengthInplaceLob()`.

Reply:
And handler in `ValueLobDb` may be null. I think that for `ValueLobDb` presence of value in `small` field can be used instead, but with such check I need to rewrite unit test to cover all branches of code.

Reply:
meh, in that case just leave it alone, that is not worth the complication for such a seldom used code path

Reply:
I already wrote a check for `ValueLobDb.small`. This check is fast and reliable. Adjusted unit test now needs to initialize an empty in-memory database to use it as backend for large lobs, but it does not matter.

For archaic `ValueLob` streams are always created in either way, so optimized code path is not very useful with them.